src/resources/views/crud/inc/form_fields_script.blade.php

```diff
@@ -0,0 +1,95 @@
+     * javascript manipulations, and makes it easy to do custom stuff
+     * too, by exposing the main components (name, wrapper, input).
+     */
+    class CrudField {
+        constructor(fieldName) {
+            this.name = fieldName;
+            this.wrapper = $("[bp-field-name="+ this.name +"]");
+            this.input = $("[bp-field-main-input][name="+ this.name +"]");
+            // if no bp-field-main-input has been declared in the field itself,
+            // assume it's the first input in that wrapper, whatever it is
+            if (this.input.length == 0) {
+                this.input = $("[bp-field-name="+ this.name +"] input, [bp-field-name="+ this.name +"] textarea, [bp-field-name="+ this.name +"] select").first();
+            }
+            this.value = this.input.val();
+        }
+
+        change(closure) {
+            this.input.change(function(event) {
+                var fieldWrapper = $(this).parent('[bp-field-wrapper=true]');
+                var fieldName = fieldWrapper.attr('bp-field-name');
+                var fieldType = fieldWrapper.attr('bp-field-type');
+                var fieldValue = $(this).val();
+
+                // console.log('Changed field ' + fieldName + ' (type '+ fieldType + '), value is now ' + fieldValue);
+                closure(event, fieldValue);
+            }).change();
+
+            return this;
+        }
+
+        onChange(closure) {
+            this.change(closure);
+        }
+
+        hide(e) {
+            this.wrapper.hide();
+            return this;
+        }
+
+        show(e) {
+            this.wrapper.show();
+            return this;
+        }
+
+        enable(e) {
+            this.input.removeAttr('disabled');
+            return this;
+        }
+
+        disable(e) {
+            this.input.attr('disabled', 'disabled');
+            return this;
+        }
+
+        require(e) {
+            this.wrapper.removeClass('required').addClass('required');
+            return this;
+        }
+
+        unrequire(e) {
+            this.wrapper.removeClass('required');
+            return this;
+        }
+
+        check(e) {
+            console.log(this.wrapper.find('input[type=checkbox]'));
+            this.wrapper.find('input[type=checkbox]').prop('checked', true).trigger('change');
+            return this;
+        }
+
+        uncheck(e) {
+            this.wrapper.find('input[type=checkbox]').prop('checked', false).trigger('change');
+            return this;
+        }
+    }
+
+    /**
+     * Window functions that help the developer easily select one or more fields.
+     */
+    window.crud = {
+        field: function(fieldName) {
+            return new CrudField(fieldName);
+        },
+        fields: function(fieldNamesArray) {
+            return fieldNamesArray.map(function(fieldName) {
+                return new CrudField(fieldName);
+            });
+        }
+    }
+</script>
```

Comment:
I'm gonna say this, because if I don't, I know Antonio will 😅  Yes, we can _easily_ eliminate jQuery from this... I know! It was just easier for me to do it in jQuery 😀

Reply:
Also, if we _do_ remove it, then the selectors will NOT be jQuery objects, so you won't be able to do stuff on them with jQuery, you'd have to wrap them... so... yeah... food for thought.